tests/unit/lms/services/application_instance_test.py

```diff
@@ -351,6 +351,7 @@ def with_application_instances_for_search(self):
                     lti_registration=registration,
                     tool_consumer_instance_guid=Faker("hexify", text="^" * 32),
                     deployment_id=Faker("hexify", text="^" * 8),
+                    name=Faker("hexify", text="^" * 6),
```

Comment:
To avoid duplicates/if you want a field to be unique you should use sequences, not random values, since random values will sometimes produce duplicates resulting in flakey tests:

```suggestion
                    name=Sequence(lambda n: f"Application Instance {n}"),
```

Might want to do the same to some of the other fields in this factory: it'll sometimes produce duplicate `tool_consumer_instance_guid`'s and `deployment_id`'s as well.

Reply:
Both times we seen this failure the random company name was: "Taylor LLC". Faker("company") was probably not meant to be used in a unique context.

I don't think a 6 digit random value would have collisions often but the sequential value is better anyway :+1: